_posts/2021-09-23-reassign-partitions.md

```diff
@@ -0,0 +1,282 @@
+---
+layout: post
+title:  "Reassign partitions in Kafka Cluster"
+date: 2021-09-23
+author: shubham_rawat
+---
+
+Scaling Kafka Brokers, removing old brokers, and increasing the number of topics and partitions are all pretty common tasks for Kafka users.
+All these cases can cause the load on the brokers to get unbalanced. The most convenient way to manage an unbalanced cluster is to reassign the partitions among all the brokers.
+In Strimzi, this is easily done with the help of the partition reassignment tool: `kafka-reassign-partitions.sh`.
+
+<!--more-->
+
+## Partition Reassignment tool
+
+Within a broker pod, the `kafka-reassign-partitions.sh` tool allows you to reassign partitions to different brokers.
+
+It has three different modes:
+
+| Mode | Description |
+| :-----------: | ------------- |
+| `generate`  | Takes a set of topics and brokers and generates a reassignment JSON file which will result in the partitions of those                     topics being assigned to those brokers. Because this operates on whole topics, it cannot be used when you only want to reassign some partitions for some topics.|
+| `execute` | Takes a reassignment JSON file and applies it to the partitions and brokers in the cluster. Brokers that gain partitions                 as a result become followers of the partition leader. For a given partition, once the new broker has caught up and joined the ISR (in-sync replicas) the old broker will stop being a follower and will delete its replica. |
```

Comment:
Same as above, why the long space in the middle?

Reply:
The `gain` sounds a a bit weird here to me. You should probably use some longer description which explains what exactly does it mean.